peekingduck/nodes/output/media_writer.py

```diff
@@ -72,13 +78,28 @@ def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
             if self.writer:  # images automatically releases writer
                 self.writer.release()
             return {}
+        self._input_type = self._detect_input_type(inputs["filename"])
+        # init
+        if self.output_filename is not None:
+            self._output_type = self._detect_input_type(self.output_filename)
         if not self._file_name:
+            self._file_name = inputs["filename"]
+
             self._prepare_writer(
-                inputs["filename"], inputs["img"], inputs["saved_video_fps"]
+                inputs["filename"],
+                inputs["img"],
+                inputs["saved_video_fps"],
+                self.output_filename,
```

Comment:
Lines 85-93 wraps the same logic as lines 94-103, so the former can be removed.
Verify this with your own testing.

Reply:
Remove duplicate code and test successfully.